mempool/reactor.go

```diff
@@ -136,7 +155,7 @@ func (memR *Reactor) broadcastTxRoutine(peer p2p.Peer) {
 
 	for {
 		// In case of both next.NextWaitChan() and peer.Quit() are variable at the same time
-		if !memR.IsRunning() || !peer.IsRunning() {
+		if !memR.IsRunning() || !peer.IsRunning() || memR.WaitSync() {
```

Comment:
This addition causes this routine to be interrupted (and never restarted) if a peer is added while the node is still syncing. This will probably break the protocol.